lib/http/node.js

```diff
@@ -76,17 +76,28 @@ AWS.NodeHttpClient = AWS.util.inherit({
 
   writeBody: function writeBody(stream, httpRequest) {
     var body = httpRequest.body;
-
-    if (body && WritableStream && ReadableStream) { // progress support
-      if (!(body instanceof Stream)) body = AWS.util.buffer.toStream(body);
-      body.pipe(this.progressStream(stream, httpRequest));
-    }
+    var totalBytes = parseInt(httpRequest.headers['Content-Length'], 10);
 
     if (body instanceof Stream) {
-      body.pipe(stream);
+      // For progress support of streaming content -
+      // pipe the data through a transform stream to emit 'sendProgress' events
+      body.pipe(this.progressStream(stream, totalBytes)).pipe(stream);
```

Comment:
Just one last change! Since we still have users running on node 0.8, we should only pipe into the progress stream if `typeof TransformStream !== undefined`.